test/library/packages/Itertools/Itertools.chpl

```diff
@@ -105,4 +104,84 @@ module Itertools {
 
     for working_iters do yield arg;
   }
+
+
+
+  /*
+    Returns elements from an iterable over and over again, a specified
+    number of times.
+
+
+    :arg arg: The iterable whose elements are to be returned
+    :type arg: `?`
+
+    :arg times: The number of times to iterate through the iterable
+    (i.e. number of times each element is to be returned)
+    :type times: `int`
+
+    :yields: Elements of the iterable ``times`` times
+
+    :throws: ``IllegalArgumentError`` on parallel infinite iteration (see below)
+
+
+    If the argument ``times`` has the value 0, it will return each element of
+    the iterable an infinite number of times.
+
+    This iterator can be called in serial and parallel zippered contexts.
+
+    .. note::
+      This iterator is not suitable for parallel infinite iteration i.e.
+      avoid using zippered, ``forall``, or ``coforall`` loops with the
+      ``times`` argument set to 0.
+  */
+
+  // Serial iterator
+
+  iter cycle(arg, times = 0) {
+    if times == 0 then
+      for 0.. do
+        for element in arg do
+          yield element;
+    else
+      for 0..#times do
+        for element in arg do
+          yield element;
+  }
+
+  use RangeChunk;
+
+  // Parallel iterator - Leader
+
+  pragma "no doc"
+  iter cycle(param tag: iterKind, arg, times = 0) throws
+      where tag == iterKind.leader {
+
+    var numTasks = if dataParTasksPerLocale > 0 then dataParTasksPerLocale
+                                                else here.maxTaskPar;
+
+    if numTasks > times then numTasks = times;
+
+    if times == 0 then
+      throw new owned IllegalArgumentError(
+          "infinite iteration not supported for parallel loops");
+    else
+      coforall tid in 0..#numTasks {
+        const working_iters = chunk(0..#times, numTasks, tid);
```

Comment:
Just caught this - it would be more Chapel-like to use camelCase: `workingIters`. Same applies to follower iterator.